targets/TARGET_STM/trng_api.c

```diff
@@ -37,6 +37,15 @@ void trng_init(trng_t *obj)
         error("Only 1 RNG instance supported\r\n");
     }
 
+#if defined(TARGET_STM32L4)
+    RCC_PeriphCLKInitTypeDef PeriphClkInitStruct;
+
+    /*Select PLLQ output as RNG clock source */
+    PeriphClkInitStruct.PeriphClockSelection = RCC_PERIPHCLK_RNG;
+    PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_PLL;
```

Comment:
PeriphClkInitStruct.RngClockSelection = RCC_RNGCLKSOURCE_MSI;

Will resolve the issue.

Reply:
@jeromecoutant  I tried this, but it will produce the same error as if the original fix.

Reply:
I see, clock configuration is different for MTB_ADV_WISE_1570 than for NUCLEO_L486RG.

Reply:
As previously was mentioned, should MTB have own system clock config for this RNG selection?

Reply:
Could you check my commit in https://github.com/jeromecoutant/mbed/tree/PR_L4TRNG ?

Reply:
@jeromecoutant  that works for me

Reply:
@jeromecoutant new PR  and we close this revert?